### PR TITLE
Make shovel use the tool system

### PR DESCRIPTION
**Pull request by PsyCommando**
## Description of changes
Relies on #2483 for the flora interaction stuff.

* Makes the shovel into a tool, and change all shovel interactions to use the tool system.
* Added digging sound. CC0 and details in the file.
* Pits can be filled up when empty to delete them.

## Why and what will this PR improve
Transition old code using type checks to use the tool archetype instead.

### CODE REVIEW

**Review by MistakeNot4892 — CHANGES_REQUESTED** [review]
Code looks good, just fix up the linter fails and we're all set.

**Comment by PsyCommando** [comment]
> Code looks good, just fix up the linter fails and we're all set.

The fail is just because the PR's dependency isn't in yet.

**Comment by PsyCommando** [comment]
Should be good now that the dependency got merged.

File: code/modules/mining/mine_items.dm [hunk 1]
```diff
@@ -175,6 +179,12 @@
 	throwforce = 7
 	w_class = ITEM_SIZE_SMALL
 
+/obj/item/shovel/spade/Initialize(ml, material_key)
```
Comment (MistakeNot4892) [inline code/modules/mining/mine_items.dm]: Would be preferable to have a var on /shovel containing the tool quality, pass that in on /shovel Initialize, then change it on /spade.

**Comment by MistakeNot4892** [comment]
Other than minor change above to avoid the double override, this code looks fine.

File: code/modules/mining/mine_items.dm (hunk 1 above)
Comment (PsyCommando) [reply code/modules/mining/mine_items.dm]: I can do that.

**Review by MistakeNot4892 — APPROVED** [review]

**Pull request merged**